framework/src/modules/chain/transport/transport.js

```diff
@@ -354,15 +355,48 @@ class Transport {
 	 * @todo Add @returns tag
 	 * @todo Add description of the function
 	 */
-	async getTransactions() {
-		const transactions = this.transactionPoolModule.getMergedTransactionList(
-			true,
-			this.constants.maxSharedTransactions,
-		);
+	async getTransactions(ids) {
+		if (!(ids && Array.isArray(ids) && ids.length)) {
+			return {
+				success: true,
+				transactions: this.transactionPoolModule.getMergedTransactionList(
+					true,
+					this.constants.maxSharedTransactions,
+				),
+			};
+		}
+
+		const transactionsFromQueues = [];
+		const idsNotInPool = [];
+
+		for (const id of ids) {
+			// Check if any transaction is in the queues.
+			const transactionInPool = this.transactionPoolModule.findInTransactionPool(
+				id,
+			);
+
+			if (transactionInPool) {
+				transactionsFromQueues.push(transactionInPool.toJSON());
+			} else {
+				idsNotInPool.push(id);
+			}
+		}
+
+		if (idsNotInPool.length) {
+			// Check if any transaction that was not in the queues, is in the database instead.
+			const transactionsFromDatabase = await this.storage.entities.Transaction.get(
+				{ id_in: idsNotInPool },
+			);
```

Comment:
This call will return maximum 10 transactions as the default `limit` for Transaction entity is 10.

In order to overcome the limit the option `{ limit: null }` needs to be set but I reckon we need some limit here to avoid peers asking for a huge number of transactions.